File: ClipboardZanager/Sources/ClipboardZanager.Shared/Core/Requires.cs
using System;
using ClipboardZanager.Shared.Exceptions;

namespace ClipboardZanager.Shared.Core
{
    /// <summary>
    /// Provides a set of methods that can be used to validate data
    /// </summary>
    public static class Requires
    {
        /// <summary>
        /// Throws an exception if the specified parameter's value is null.
        /// </summary>
        /// <param name="value">The value to test.</param>
        /// <param name="parameterName">The name of the parameter to include in any thrown exception.</param>
        public static void NotNull(object value, string parameterName)
        {
            if (value == null)
            {
                throw new NotNullRequiredException(parameterName);
            }
        }

        /// <summary>
        /// Throws an exception if the specified parameter's value is null or empty.
        /// </summary>
        /// <param name="value">The value to test.</param>
        /// <param name="parameterName">The name of the parameter to include in any thrown exception.</param>
        public static void NotNullOrEmpty(string value, string parameterName)
        {
            if (string.IsNullOrEmpty(value))
            {
                throw new NotNullOrEmptyRequiredException(parameterName);
            }
        }

        /// <summary>
        /// Throws an exception if the specified parameter's value is null, empty, or whitespace.
        /// </summary>
        /// <param name="value">The value to test.</param>
        /// <param name="parameterName">The name of the parameter to include in any thrown exception.</param>
        public static void NotNullOrWhiteSpace(string value, string parameterName)
        {
            if (string.IsNullOrWhiteSpace(value))
            {
                throw new NotNullOrWhiteSpaceRequiredException(parameterName);
            }
        }

        /// <summary>
        /// Throws an exception if the specified parameter's value is not false.
        /// </summary>
        /// <param name="value">The value to test.</param>
        public static void IsFalse(bool value)
        {
            if (value)
            {
                throw new IsFalseRequiredException();
            }
        }

        /// <summary>
        /// Throws an exception if the specified parameter's value is not true.
        /// </summary>
        /// <param name="value">The value to test.</param>
        public static void IsTrue(bool value)
        {
            if (!value)
            {
                throw new IsTrueRequiredException();
            }
        }

        /// <summary>
        /// Throws an exception if the specified parameter's value is > 1.
        /// </summary>
        /// <param name="hresult">The value to test.</param>
        public static void VerifySucceeded(int hresult)
        {
            if (hresult > 1)
            {
                throw new InvalidOperationException("Failed with HRESULT: " + hresult.ToString("X"));
            }
        }

        /// <summary>
        /// Returns a boolean that defines whether an operation succeeded.
        /// </summary>
        /// <param name="hresult">The value to test.</param>
        /// <returns>Returns True if it succeeded</returns>
        public static bool Succeeded(int hresult)
        {
            return hresult >= 0;
        }

        /// <summary>
        /// Returns a boolean that defines whether an operation failed.
        /// </summary>
        /// <param name="hresult">The value to test.</param>
        /// <returns>Returns False if it failed</returns>
        public static bool Failed(int hresult)
        {
            return hresult < 0;
        }
    }
}

File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/componentmodel/core/Requires.java
package com.etiennebaudoux.clipboardzanager.componentmodel.core;

import com.etiennebaudoux.clipboardzanager.componentmodel.exceptions.IsFalseRequiredException;
import com.etiennebaudoux.clipboardzanager.componentmodel.exceptions.IsTrueRequiredException;
import com.etiennebaudoux.clipboardzanager.componentmodel.exceptions.NotNullOrEmptyRequiredException;
import com.etiennebaudoux.clipboardzanager.componentmodel.exceptions.NotNullOrWhiteSpaceRequiredException;
import com.etiennebaudoux.clipboardzanager.componentmodel.exceptions.NotNullRequiredException;

/**
 * Provides a set of methods that can be used to validate data.
 */

public final class Requires {
    /**
     * Throws an exception if the specified parameter's value is null.
     *
     * @param value         The value to test.
     * @param parameterName The name of the parameter to include in any thrown exception.
     */
    public static void notNull(Object value, String parameterName) {
        if (value == null) {
            throw new NotNullRequiredException(parameterName);
        }
    }

    /**
     * Throws an exception if the specified parameter's value is null or empty.
     *
     * @param value         The value to test.
     * @param parameterName The name of the parameter to include in any thrown exception.
     */
    public static void notNullOrEmpty(String value, String parameterName) {
        if (StringUtils.isNullOrEmpty(value)) {
            throw new NotNullOrEmptyRequiredException(parameterName);
        }
    }

    /**
     * Throws an exception if the specified parameter's value is null, empty, or whitespace.
     *
     * @param value         The value to test.
     * @param parameterName The name of the parameter to include in any thrown exception.
     */
    public static void notNullOrWhiteSpace(String value, String parameterName) {
        if (StringUtils.isEmptyOrWhiteSpace(value)) {
            throw new NotNullOrWhiteSpaceRequiredException(parameterName);
        }
    }

    /**
     * Throws an exception if the specified parameter's value is not false.
     *
     * @param value The value to test.
     */
    public static void isFalse(boolean value) {
        if (value) {
            throw new IsFalseRequiredException();
        }
    }

    /**
     * Throws an exception if the specified parameter's value is not true.
     *
     * @param value The value to test.
     */
    public static void isTrue(boolean value) {
        if (!value) {
            throw new IsTrueRequiredException();
        }
    }
}
